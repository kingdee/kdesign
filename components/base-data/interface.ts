import { tuple } from '../_utils/type'

export const Modes = tuple('multiple', 'single')
export type Mode = typeof Modes[number]

export type SelectValue = string | string[] | number | number[] | IOptionsProps[] | null

export interface IAdvancedSelectorProps<T = SelectValue> {
  prefixCls?: string // 样式前缀
  className?: string // 样式名
  children?: React.ReactNode
  style?: React.CSSProperties // 内联样式
  disabled?: boolean // 禁用
  dropdownStyle?: React.CSSProperties // 下拉样式
  notFoundContent?: React.ReactNode // 列表为空时文本
  placeholder?: string // 选择框默认文字
  showFrequent?: boolean // 是否显示历史记录与我的收藏面板
  showCollectIcon?: boolean // 是否显示收藏按钮
  showDetailIcon?: boolean // 是否显示详情按钮
  suffixIcon?: React.ReactNode // 选择框后缀图标
  mode?: Mode // 模式
  value?: IOptionsProps[] | IOptionsProps // 指定选中的条目
  columns?: [] // 下拉列表的展示列
  options?: [] // 下拉列表的数据
  size?: number // 下拉列表显示数量
  optionLabelProp?: string // 回填到选择框的属性
  dropdownFooterRender?: React.ReactNode // 自定义下拉框footer内容
  visible?: boolean
  collectList?: [] // 收藏列表
  historyList?: [] // 我的历史
  loading?: boolean // 加载中状态
  searchField?: string // 搜索字段 默认 all（全部字段）
  onChange?: () => void // 选中option调用此函数
  onSearch?: (val: string) => void // input的value变化时调用
  onSelect?: (value: T extends (infer I)[] ? I : T, option: React.ReactElement<any>) => void
  onShowDetail?: (value: IOptionsProps[]) => void
  onShowMore?: () => void
  onCollect?: (flag: boolean, value: IOptionsProps) => void
}

export interface IIndxPosListProps {
  end: number
  start: number
}

export interface IColumnsProps {
  title?: string
  key: string
}

export interface IOptionsProps {
  value?: string | number
  label?: string
}

export interface ISearchInfoProps {
  searchValue?: string
  previousInputValue: string
  previousEditValue?: string
  searchIndex?: number
  selectionStart?: number
  deleteEndIndx: number
  editOptions?: any[]
}
