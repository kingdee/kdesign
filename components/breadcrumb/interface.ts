import { ReactNode } from 'react'
import { DropDownProps } from '../dropdown'
import { tuple } from '../_utils/type'

export const ColorModelTypes = tuple('emphasize', 'weaken')
export type ColorModelType = typeof ColorModelTypes[number]

export interface IBreadcrumbProps {
  style?: Record<string, unknown> // 内联样式
  className?: string // 类名
  colorModel?: ColorModelType // 颜色模式，默认是 'emphasize'
  prefixCls?: string // 样式前缀
  items: Array<IBreadcrumbItem> // 面包屑导航的各项，支持传入 ReactNode 类型
  separator?: ReactNode // 分隔符，默认是 '/'
  children?: ReactNode // 子元素
  onItemClick?: (item: IBreadcrumbItem, index: number) => void // 点击面包屑导航的回调函数
}

export interface IBreadcrumbItems {
  item: IBreadcrumbItem
  index: number
  separator?: ReactNode
  openEllipsis?: boolean
  onItemClick?: (item: IBreadcrumbItem, index: number) => void // 点击面包屑导航的回调函数
}

export interface IBreadcrumbItem {
  className?: string // 类名
  title?: ReactNode
  dropdownProps?: DropDownProps
  href?: string // 链接地址，不能和path共用
  path?: string // 拼接地址，不能和href共用
  icon?: ReactNode
}

export interface IItemsWidth {
  width: number
  index: number
}
