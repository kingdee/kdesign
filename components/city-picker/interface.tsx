import * as React from 'react'
import { tuple } from '../_utils/type'

export type City = {
  id?: string | number
  country?: string
  province?: string
  name?: string
  [key: string]: any
}

export const BorderTypes = tuple('none', 'underline', 'bordered')
export type BorderType = typeof BorderTypes[number]

export const Sizes = tuple('large', 'middle', 'small') // 选择框大小
export type Size = typeof Sizes[number]

export type Type = 'domestic' | 'foreign'

export type CityList = Array<City>

export interface CityPickerProps {
  prefixCls?: string // 样式前缀
  size?: Size // 尺寸
  borderType?: BorderType // 边框类型
  notFoundContent?: React.ReactNode // 列表为空时文本
  showArrow?: boolean // 显示下拉小箭头
  defaultOpen?: boolean // 默认是否展开下拉列表
  allowClear?: boolean // 允许清除内容
  placeholder?: React.ReactNode // 选择框默认文字
  type?: Type
  loading?: boolean
  domesticList?: CityList
  foreignList?: CityList
  commonList?: CityList
  className?: string
  disabled?: boolean
  style?: React.CSSProperties
  title?: string
  clearIcon?: React.ReactNode // 清除图标
  suffixIcon?: React.ReactNode // 选择框后缀图标
  optionHighlightProps?: string // 搜索时 根据该属性值高亮城市名
  value?: string | number
  defaultValue?: string | number
  description?: React.ReactNode
  showDescription?: boolean
  onChange?: (value: string | number, city: City) => void
  onBlur?: (e: React.ChangeEvent<HTMLSpanElement>) => void
  onFocus?: () => void
  onSearch?: (value: string) => void
  onClear?: () => void
  itemRender?: (city: City) => React.ReactNode
  onTabPaneChange?: (id: Type) => void
}

export interface ICityPickerOptionProps {
  disabled?: boolean
  city?: City
  value?: string | number
  children?: React.ReactNode
  className?: string
  style?: React.CSSProperties
  onChangeSelect?: (city: City | null) => void
  renderCityInfo?: (city: City, flag?: boolean, symbol?: string) => React.ReactNode
  itemRender?: (city: City) => React.ReactNode
}
