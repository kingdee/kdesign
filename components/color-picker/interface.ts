import { tuple } from '../_utils/type'
import { PopperProps } from '../_utils/usePopper'

export const ColorTypes = tuple('HEX', 'HSB', 'RGB', 'HSL', 'themeColor')
export type ColorType = typeof ColorTypes[number]
export interface IColorTypesObj {
  type: ColorType
  value: string
}

export const BorderTypes = tuple('underline', 'bordered')
export type BorderType = typeof BorderTypes[number]
export interface IColorPickerProps extends PopperProps {
  // 设置颜色选择器的值，可以是十六进制颜色值，RGB颜色值，HSL颜色值，或者颜色名称
  value: string
  // 设置颜色选择器的默认值，可以是十六进制颜色值，RGB颜色值，HSL颜色值，或者颜色名称
  defaultValue: string
  // 选择器输入框类名
  className: string
  // 选择输入框样式
  style: React.CSSProperties
  // 输入框类型
  borderType: BorderType
  // 输入框右侧的图标
  suffixIcon: (rgbColor: string) => React.ReactNode
  // 功能色名称
  functionalColorName: string
  // 功能色色值
  functionalColor: string
  // 功能色开关的名称
  switchName: { name: string; internationalName?: string }
  // 是否跟随功能色,定义了功能色才能控制显示跟随主题色开关
  showSwitch: boolean
  // 是否展示色彩格式转换框
  showColorTransfer: boolean
  // 是否展示系统预设颜色
  showPresetColor: boolean
  // 占位字符
  placeholder: string
  // 用户自定义的预设颜色
  presetColor: string[]
  // 是否展示拾色容器
  showColorPickerBox: { showBox: boolean; showHue: boolean; showOpacity: boolean }
  // 颜色选择面板显隐配置
  defaultOpen: boolean
  // 当颜色选择器的值发生变化时触发的回调函数
  onChange: (inputValue: string, formatColorArr: Array<IColorTypesObj>) => void
}

export interface IColorPickerPanelProps {
  alpha: number
  alphaNoVerifyVal: string
  clickedColorIndex?: number
  colTypeArr: Array<IColorTypesObj>
  correctColorValue: string
  currentColorType: IColorTypesObj['type']
  functionalColor?: string
  functionalColorName: string
  switchName: { name: string; internationalName?: string }
  inputRef: any
  isFollow: boolean
  presetColor?: string[]
  showSwitch?: boolean
  showColorTransfer?: boolean
  showPresetColor?: boolean
  showColorPickerBox?: { showBox?: boolean; showHue?: boolean; showOpacity?: boolean }
  showColorPickerPanel: boolean
  value: string
  setCurrentColorType: (currentColorType: IColorTypesObj['type']) => void
  setColTypeArr: (colTypeArr: Array<IColorTypesObj>) => void
  setClickedColorIndex: (clickedColorIndex: number) => void
  setIsFollow: (isFollow: boolean) => void
  setAlphaNoVerifyVal: (NoVerifyVal: string) => void
  setAlpha: (alpha: number) => void
  setCorrectColorValue: (colorValue: string) => void
  setInputColorValue: (colorValue: string) => void
  setShowPanel: (showPanel: boolean) => void
  onChange?: (inputValue: string, formatColorArr: Array<IColorTypesObj>) => void
}
