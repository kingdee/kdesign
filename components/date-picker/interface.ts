import { BorderType, InputSiteType } from '../input/input'
import { DisabledDataProps } from './date-picker'

export type WeekTitleType = [string, string, string, string, string, string, string]
export type MonthTitleType = [
  string,
  string,
  string,
  string,
  string,
  string,
  string,
  string,
  string,
  string,
  string,
  string,
]

export interface Locale {
  // locale: string
  'DatePicker.placeholder': string
  'DatePicker.yearPlaceholder': string
  'DatePicker.quarterPlaceholder': string
  'DatePicker.monthPlaceholder': string
  'DatePicker.weekPlaceholder': string
  'DatePicker.timePlaceholder': string
  'DatePicker.now': string
  'DatePicker.confrim': string
  'DatePicker.today': string
  'DatePicker.year': string
  'DatePicker.month': string
  'DatePicker.weekTitle': WeekTitleType
  'DatePicker.monthTitle': MonthTitleType
  'DatePicker.rangePlaceholder': [string, string]
  'DatePicker.rangeYearPlaceholder': [string, string]
  'DatePicker.rangeMonthPlaceholder': [string, string]
  'DatePicker.rangeWeekPlaceholder': [string, string]
  'DatePicker.rangeQuarterPlaceholder': [string, string]
  'DatePicker.rangeTimePlaceholder': [string, string]
}

export interface InnerLocale {
  locale: string
  placeholder: string
  yearPlaceholder: string
  quarterPlaceholder: string
  monthPlaceholder: string
  weekPlaceholder: string
  timePlaceholder: string
  now: string
  confrim: string
  today: string
  year: string
  month: string
  weekTitle: WeekTitleType
  monthTitle: MonthTitleType
  rangePlaceholder: [string, string]
  rangeYearPlaceholder: [string, string]
  rangeMonthPlaceholder: [string, string]
  rangeWeekPlaceholder: [string, string]
  rangeQuarterPlaceholder: [string, string]
  rangeTimePlaceholder: [string, string]
}

export type InnerLocaleKey = keyof InnerLocale

export type PanelMode = 'time' | 'date' | 'week' | 'month' | 'quarter' | 'year' | 'decade'

export type PickerMode = Exclude<PanelMode, 'datetime' | 'decade'>

export type DateType = Date

export type CustomFormat = (value: DateType) => string

export type DateFormat = string

export type DateFormats = Array<string>

export type NullableDateType = DateType | null | undefined

export type OnSelect = (value: DateType, type: 'key' | 'mouse' | 'submit') => void
export interface DisabledTimes {
  disabledHours?: () => number[]
  disabledMinutes?: (hour: number) => number[]
  disabledSeconds?: (hour: number, minute: number) => number[]
}

export type DisabledTime = (date: DateType | null) => DisabledTimes

export type OnPanelChange = (value: DateType, mode: PanelMode) => void

export type EventValue = DateType | null
export type RangeValue = [EventValue, EventValue] | null

export type RangeList = {
  label: string
  onClick: () => void
  onMouseEnter: () => void
  onMouseLeave: () => void
}[]

export interface Components {
  button?: React.ComponentType | string
  rangeItem?: React.ComponentType | string
}

export interface PanelSharedProps {
  value?: NullableDateType
  defaultPickerValue?: DateType
  locale: InnerLocale
  disabledDate?: DisabledDataProps

  prevIcon?: React.ReactNode
  nextIcon?: React.ReactNode
  superPrevIcon?: React.ReactNode
  superNextIcon?: React.ReactNode

  // 面板操作（焦点操作、键盘操作）next version
  // operationRef: React.MutableRefObject<PanelRefProps>

  // onPanelChange: (mode: PanelMode | null, viewValue: DateType) => void
}

export type DateRender = (currentDate: DateType, today: DateType) => React.ReactNode

export type CellRender = (currentDate: DateType) => React.ReactNode

export type RangeType = 'start' | 'end'

export interface RangeInfo {
  range: RangeType
}

export type RangeDateRender = (currentDate: DateType, today: DateType, info: RangeInfo) => React.ReactNode

export interface SharedTimeProps extends DisabledTimes {
  format?: string
  showNow?: boolean
  showHour?: boolean
  showMinute?: boolean
  showSecond?: boolean
  use12Hours?: boolean
  hourStep?: number
  minuteStep?: number
  secondStep?: number
  hideDisabledOptions?: boolean
  defaultValue?: DateType
}

export interface OutInputCommonProps {
  format: DateFormat
  picker: PickerMode
  id?: string
  autoFocus?: boolean
  allowClear?: boolean
  className?: string
  borderType?: BorderType
  style?: React.CSSProperties
  size?: InputSiteType
  readOnly?: boolean
  suffixIcon?: React.ReactNode
  clearIcon?: React.ReactNode

  onFocus?: React.FocusEventHandler<HTMLInputElement>
  onBlur?: React.FocusEventHandler<HTMLInputElement>
  onMouseDown?: React.MouseEventHandler<HTMLDivElement>
  onMouseUp?: React.MouseEventHandler<HTMLDivElement>
  onMouseEnter?: React.MouseEventHandler<HTMLDivElement>
  onMouseLeave?: React.MouseEventHandler<HTMLDivElement>
  onClick?: React.MouseEventHandler<HTMLDivElement>
  onContextMenu?: React.MouseEventHandler<HTMLDivElement>
}

export interface InputCommonProps {
  panelDivRef: React.RefObject<HTMLDivElement>
  needConfirmButton: boolean
  open: boolean
  prefixCls: string
  locale: InnerLocale
  dataOrAriaProps?: Record<string, any>
  status?: 'error'
}

export interface OutPopperProps {
  dropdownClassName?: string
  popupStyle?: React.CSSProperties
  showArrow?: boolean
}

export interface TimeUnit {
  label: React.ReactText
  value: number
  disabled: boolean
}
