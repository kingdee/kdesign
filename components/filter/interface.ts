import React from 'react'
import { Menu } from '../dropdown/menu'
import { IQuickSearchProps } from '../search'
import { DirectionType } from '../config-provider/ConfigContext'

type TabKey = 'condition' | 'scheme'

export type TFilterValue = {
  [key: string]: string[] | IScheme
}

export type TOption = {
  value?: string
  label?: string
  component?: React.ReactNode
}

export interface ICondition {
  key: string
  label: string
  options: TOption[]
  required?: boolean
  multiple?: boolean
  addition?: React.ReactNode
}

export interface IField {
  name: string
  label: string
  required?: boolean
  operators: React.ReactElement
  assignOptions: Record<string, React.ReactElement>
}

export interface ISchemeItem {
  fieldName: string
  value: string
  operator: string
  required?: boolean
}

export interface IScheme {
  name?: string
  preset?: boolean
  items: ISchemeItem[]
}

export interface IFilterProps {
  spread?: boolean
  fields?: IField[]
  className?: string
  schemes?: IScheme[]
  value?: TFilterValue
  defaultTabKey?: TabKey
  defaultSpread?: boolean
  title?: React.ReactNode
  conditions?: ICondition[]
  search?: IQuickSearchProps
  style?: React.CSSProperties
  defaultValue?: TFilterValue
  onSchemeSave?: (scheme: IScheme) => void
  onSpreadChange?: (spread: boolean) => void
  getSchemeSettingMenu?: (scheme: IScheme) => typeof Menu
  onChange?: (value: TFilterValue, condition?: ICondition, option?: TOption) => void
}

export interface IConditionProps {
  tab: string
  prefixCls: string
  value: TFilterValue
  conditions: ICondition[]
  FilterLangMsg: Record<string, any>
  onConditionRemove: (key: string, e: React.MouseEvent) => void
  onConditionChange: (value: TFilterValue, condition?: ICondition, option?: TOption) => void
}

export interface ISchemeProps {
  tab: string
  fields: IField[]
  prefixCls: string
  schemes: IScheme[]
  direction: DirectionType
  value: { scheme: IScheme }
  FilterLangMsg: Record<string, any>
  onSchemeSave: (scheme: IScheme) => void
  onSchemeSearch: (scheme: IScheme) => void
  getSchemeSettingMenu?: (scheme: IScheme) => void
}
