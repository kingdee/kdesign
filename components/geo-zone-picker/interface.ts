import { tuple } from '../_utils/type'
export const BorderTypes = tuple('underline', 'bordered')
export type BorderType = typeof BorderTypes[number]
const TabTypes = tuple('PROVINCE', 'CITY', 'DISTRICT')
export type TabType = typeof TabTypes[number]
export interface Country {
  code: number | string
  name: string
}

export interface Region {
  code: string
  name: string
  group?: string
  children?: Region[]
}

// 声明数组对象的类型
export type RegionArray = Array<Region>

export interface RegionValue {
  countryCode?: string | number
  countryName?: string
  provinceCode?: string | number
  provinceName?: string
  cityCode?: string | number
  cityName?: string
  districtCode?: string | number
  districtName?: string
}
