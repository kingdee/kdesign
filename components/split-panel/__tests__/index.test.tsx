import React from 'react'
import { render, mount } from 'enzyme'
import { act } from 'react-dom/test-utils'
import SplitPanel, { SplitPanelModes } from '../split-panel'
import { SplitPanelMode } from '../interface'
import Icon from '../../icon'
import mountTest from '../../../tests/shared/mountTest'
import ConfigProvider from '../../config-provider/index'

const current = {
  clientWidth: 200,
  clientHeight: 200,
  getBoundingClientRect() {
    return {
      top: 0,
      left: 0,
    }
  },
}

jest.mock('react', () => {
  const mockedModule = jest.genMockFromModule('react')
  const originalModule = jest.requireActual('react')
  return Object.assign({}, mockedModule, originalModule, {
    useRef: jest.fn(() =>
      Object.defineProperty({}, 'current', {
        get: () => current,
        set: () => current,
      }),
    ),
  })
})

describe('SplitPanel', () => {
  // 1.mount test
  mountTest(SplitPanel)

  SplitPanelModes.forEach((mode) => {
    mountTest(() => <SplitPanel mode={mode} />)
  })

  // 2.render test
  it('renders correctly', () => {
    SplitPanelModes.forEach((mode) => {
      expect(render(<SplitPanel mode={mode} />)).toMatchSnapshot()
    })
  })

  // 3.render no child without errors
  it('render no child without errors', () => {
    expect(mount(<SplitPanel></SplitPanel>)).toMatchSnapshot()
  })

  // 4.render null or undefined without errors
  it('render null or undefined without errors', () => {
    expect(
      mount(
        <SplitPanel>
          {null}
          {undefined}
        </SplitPanel>,
      ),
    ).toMatchSnapshot()
  })
  // 5.test warn
  // warns if mode is wrong
  it('warns if type is wrong', () => {
    const mockWarn = jest.fn()
    jest.spyOn(console, 'warn').mockImplementation(mockWarn)
    const mode = 'any mode' as SplitPanelMode
    render(<SplitPanel mode={mode} />)
    expect(mockWarn).toHaveBeenCalledTimes(1)
    expect(mockWarn.mock.calls[0][0]).toMatch(
      "Warning: [kdesign]-split-panel: cannot found split-panel mode 'any mode'",
    )
  })

  // warns if defaultSplit is wrong
  it('warns if defaultSplit is wrong', () => {
    const mockWarn = jest.fn()
    jest.spyOn(console, 'warn').mockImplementation(mockWarn)
    const defaultSplit = 'abc'
    render(<SplitPanel defaultSplit={defaultSplit} />)
    expect(mockWarn).toHaveBeenCalledTimes(1)
    expect(mockWarn.mock.calls[0][0]).toMatch("Warning: [kdesign]-split-panel: incorrect prop 'defaultSplit'")
  })

  // 6.displayName
  it('should have displayName static property', () => {
    const wrapper = mount(<SplitPanel></SplitPanel>)
    expect((wrapper.type() as any).displayName).toBe('SplitPanel')
  })

  // 7.className state
  it('should class use right', () => {
    const wrapper = mount(<SplitPanel className="my-test" style={{ color: 'red' }} data-test="test"></SplitPanel>)
    expect(wrapper.prop('data-test')).toEqual('test')
    expect(wrapper.find('.kd-split-wrapper')).toHaveClassName('my-test')
    expect(wrapper.find('.kd-split-wrapper')).toHaveStyle('color', 'red')
  })

  // 8.test defaultSplit compute correctly
  it('test defaultSplit compute correctly', () => {
    const wrapper = mount(<SplitPanel defaultSplit="50px" />)
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '75%')
    wrapper.setProps({ defaultSplit: '210px' })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '0%')
    wrapper.setProps({ defaultSplit: '-10px' })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '100%')
  })

  // 9.test compute min and max correctly
  it('test compute min and max correctly', () => {
    const wrapper = mount(<SplitPanel min={0.6} max={0.6} />)
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '60%')
    wrapper.setProps({
      min: '40px',
      max: '120px',
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '60%')
    wrapper.setProps({
      min: '120px',
      max: '40px',
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '40%')
  })

  // 10.test drag trigger line
  it('test drag trigger line', () => {
    document.documentElement.scrollLeft = 0
    document.documentElement.scrollTop = 0
    const onMoveStart = jest.fn()
    const onMoving = jest.fn()
    const onMoveEnd = jest.fn()
    const wrapper = mount(
      <SplitPanel min={0.2} max={0.2} onMoveStart={onMoveStart} onMoving={onMoving} onMoveEnd={onMoveEnd} />,
    )
    wrapper.find('.kd-split-trigger').simulate('mousedown')
    expect(onMoveStart).toHaveBeenCalledTimes(1)
    act(() => {
      const event = new MouseEvent('mousemove')
      Object.defineProperties(event, {
        pageX: {
          value: 44,
        },
        pageY: {
          value: 44,
        },
      })
      document.dispatchEvent(event)
    })
    expect(onMoving).toHaveBeenCalledTimes(1)
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '78%')
    act(() => {
      const event = new MouseEvent('mouseup')
      document.dispatchEvent(event)
    })
    expect(onMoveEnd).toHaveBeenCalledTimes(1)
  })

  // 11.test move instance is over the min or max value
  it('test move instance is over the min or max value', () => {
    document.documentElement.scrollLeft = 0
    document.documentElement.scrollTop = 0
    const onMoveStart = jest.fn()
    const onMoveEnd = jest.fn()
    const wrapper = mount(<SplitPanel min={0.2} max={0.2} onMoveStart={onMoveStart} onMoveEnd={onMoveEnd} />)
    wrapper.find('.kd-split-trigger').simulate('mousedown')
    expect(onMoveStart).toHaveBeenCalledTimes(1)
    act(() => {
      const event = new MouseEvent('mousemove')
      Object.defineProperties(event, {
        pageX: {
          value: 10,
        },
        pageY: {
          value: 10,
        },
      })
      document.dispatchEvent(event)
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '80%')
    act(() => {
      const event = new MouseEvent('mousemove')
      Object.defineProperties(event, {
        pageX: {
          value: 190,
        },
        pageY: {
          value: 190,
        },
      })
      document.dispatchEvent(event)
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '20%')
    act(() => {
      const event = new MouseEvent('mouseup')
      document.dispatchEvent(event)
    })
    expect(onMoveEnd).toHaveBeenCalledTimes(1)
  })

  // 12.Api test
  it('test fold correctly', () => {
    const onFold = jest.fn()
    const wrapper = mount(<SplitPanel min={0.2} max={0.2} onFold={onFold} />)
    const firstFoldNode = wrapper.find('.kd-split-trigger-arrow').first()
    const lastFoldNode = wrapper.find('.kd-split-trigger-arrow').last()
    act(() => {
      firstFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '100%')
    act(() => {
      lastFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '80%')
    act(() => {
      lastFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '50%')
    act(() => {
      lastFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '0%')
    act(() => {
      firstFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '20%')
    act(() => {
      firstFoldNode.simulate('click')
    })
    wrapper.update()
    expect(wrapper.find('.kd-split-panel').first().prop('style')).toHaveProperty('right', '50%')
  })

  it('test disable mode', () => {
    const onFold = jest.fn()
    const onMoveStart = jest.fn()
    const wrapper = mount(<SplitPanel disabled min={0.2} max={0.2} onFold={onFold} onMoveStart={onMoveStart} />)
    const firstFoldNode = wrapper.find('.kd-split-trigger-arrow').first()
    const lastFoldNode = wrapper.find('.kd-split-trigger-arrow').last()
    wrapper.find('.kd-split-trigger').simulate('mousedown')
    expect(onMoveStart).not.toHaveBeenCalled()
    firstFoldNode.simulate('click')
    lastFoldNode.simulate('click')
    expect(onFold).not.toHaveBeenCalled()
  })

  it('test custom lineStyle', () => {
    const lineStyle = (mode: SplitPanelMode) => {
      if (mode === 'horizontal') {
        return { width: '2px', background: '#5582f3' }
      }
    }
    const wrapper = mount(<SplitPanel lineStyle={lineStyle} />)
    expect(wrapper.find('.kd-split-trigger').prop('style')).toEqual({
      width: '2px',
      background: '#5582f3',
    })
  })

  it('test canFold is false', () => {
    const wrapper = mount(<SplitPanel canFold={false} />)
    expect(wrapper.find('.kd-split-trigger-arrow').length).toBe(0)
  })

  it('test custom foldIcons', () => {
    const foldIcons = (mode: SplitPanelMode) => {
      if (mode === 'horizontal') {
        return [<Icon type="arrow-left" key="arrow-left" />, <Icon type="arrow-right" key="arrow-right" />]
      }
    }
    const wrapper = mount(<SplitPanel foldIcons={foldIcons} />)
    expect(wrapper.find('.kdicon-arrow-left').length).toBe(1)
    expect(wrapper.find('.kdicon-arrow-right').length).toBe(1)
  })

  // 13.config provider
  it('should config use config provider', () => {
    const splitPanelConfig = {
      compDefaultProps: {
        SplitPanel: {
          mode: 'vertical',
          defaultSplit: 0.7,
        },
      },
    }
    const wrapper = mount(
      <ConfigProvider value={splitPanelConfig}>
        <SplitPanel />
      </ConfigProvider>,
    )
    expect(wrapper.find('.kd-split-vertical').exists()).toBe(true)
    expect(wrapper.find('.kd-split-trigger-con').prop('style')).toHaveProperty('top', '70%')
  })
})
