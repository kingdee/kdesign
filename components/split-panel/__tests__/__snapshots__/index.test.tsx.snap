// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitPanel render no child without errors 1`] = `
<SplitPanel>
  <div
    className="kd-split-wrapper"
  >
    <div
      className="kd-split-horizontal"
    >
      <div
        className="kd-split-panel left-panel"
        style={
          Object {
            "right": "50%",
          }
        }
      />
      <div
        className="kd-split-trigger-con can-drag"
        style={
          Object {
            "left": "50%",
          }
        }
      >
        <Memo(Trigger)
          handleMousedown={[Function]}
          hideFirstPanel={[Function]}
          hideSecondPanel={[Function]}
          mode="vertical"
          prefixCls="kd-split"
          showFoldIcons={true}
          style={Object {}}
        >
          <div
            className="kd-split-trigger kd-split-trigger-vertical"
            onMouseDown={[Function]}
            style={Object {}}
          />
          <span
            className="kd-split-trigger-arrow is-vertical left"
            onClick={[Function]}
          >
            <Icon
              type="arrow-left-solid"
            >
              <i
                className="kdicon kdicon-arrow-left-solid"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={Object {}}
              />
            </Icon>
          </span>
          <span
            className="kd-split-trigger-arrow is-vertical right"
            onClick={[Function]}
          >
            <Icon
              type="arrow-right-solid"
            >
              <i
                className="kdicon kdicon-arrow-right-solid"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={Object {}}
              />
            </Icon>
          </span>
        </Memo(Trigger)>
      </div>
      <div
        className="kd-split-panel right-panel"
        style={
          Object {
            "left": "50%",
          }
        }
      />
    </div>
  </div>
</SplitPanel>
`;

exports[`SplitPanel render null or undefined without errors 1`] = `
<SplitPanel>
  <div
    className="kd-split-wrapper"
  >
    <div
      className="kd-split-horizontal"
    >
      <div
        className="kd-split-panel left-panel"
        style={
          Object {
            "right": "50%",
          }
        }
      />
      <div
        className="kd-split-trigger-con can-drag"
        style={
          Object {
            "left": "50%",
          }
        }
      >
        <Memo(Trigger)
          handleMousedown={[Function]}
          hideFirstPanel={[Function]}
          hideSecondPanel={[Function]}
          mode="vertical"
          prefixCls="kd-split"
          showFoldIcons={true}
          style={Object {}}
        >
          <div
            className="kd-split-trigger kd-split-trigger-vertical"
            onMouseDown={[Function]}
            style={Object {}}
          />
          <span
            className="kd-split-trigger-arrow is-vertical left"
            onClick={[Function]}
          >
            <Icon
              type="arrow-left-solid"
            >
              <i
                className="kdicon kdicon-arrow-left-solid"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={Object {}}
              />
            </Icon>
          </span>
          <span
            className="kd-split-trigger-arrow is-vertical right"
            onClick={[Function]}
          >
            <Icon
              type="arrow-right-solid"
            >
              <i
                className="kdicon kdicon-arrow-right-solid"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={Object {}}
              />
            </Icon>
          </span>
        </Memo(Trigger)>
      </div>
      <div
        className="kd-split-panel right-panel"
        style={
          Object {
            "left": "50%",
          }
        }
      />
    </div>
  </div>
</SplitPanel>
`;

exports[`SplitPanel renders correctly 1`] = `
<div
  class="kd-split-wrapper"
>
  <div
    class="kd-split-horizontal"
  >
    <div
      class="kd-split-panel left-panel"
      style="right:100%"
    />
    <div
      class="kd-split-trigger-con can-drag is-active"
      style="left:0%"
    >
      <div
        class="kd-split-trigger kd-split-trigger-vertical"
      />
      <span
        class="kd-split-trigger-arrow is-vertical left"
      >
        <i
          class="kdicon kdicon-arrow-left-solid"
        />
      </span>
      <span
        class="kd-split-trigger-arrow is-vertical right"
      >
        <i
          class="kdicon kdicon-arrow-right-solid"
        />
      </span>
    </div>
    <div
      class="kd-split-panel right-panel"
      style="left:0%"
    />
  </div>
</div>
`;

exports[`SplitPanel renders correctly 2`] = `
<div
  class="kd-split-wrapper"
>
  <div
    class="kd-split-vertical"
  >
    <div
      class="kd-split-panel top-panel"
      style="bottom:100%"
    />
    <div
      class="kd-split-trigger-con can-drag is-active"
      style="top:0%"
    >
      <div
        class="kd-split-trigger kd-split-trigger-horizontal"
      />
      <span
        class="kd-split-trigger-arrow is-horizontal top"
      >
        <i
          class="kdicon kdicon-arrow-up-solid"
        />
      </span>
      <span
        class="kd-split-trigger-arrow is-horizontal bottom"
      >
        <i
          class="kdicon kdicon-arrow-down-solid"
        />
      </span>
    </div>
    <div
      class="kd-split-panel bottom-panel"
      style="top:0%"
    />
  </div>
</div>
`;
